### ISSUE TITLE
End with active goal

### ISSUE THREAD

**Issue by PlaSTyx** [coq/coq#14139]
#### Description of the problem
coqc don't warn about active goal when End (Module/Section) is parsed.

Futhermore, provided code returns
Anomaly "Uncaught exception Not_found." 

```
Module Type Ťeq.
Section With_A.
Variable A : Type.
Parameter eq : forall x y:A, Prop.
Parameter eq_refl : forall x:A, eq x x.
End With_A.
End Ťeq.
(*
Module Test.
Definition Any : True.
End Test. (* everything's OK ! Don't care about goal :) *)
*)
Module Test (Import T:Ťeq).
Definition Streicher_K_on_ (U:Type)(x : U) (P : @eq U x x -> Prop) :=
    P (@eq_refl _ x) -> forall p : @eq U x x, P p.
Definition Streicher_K_ := forall U x P, @Streicher_K_on_ U x P.

Section useless.
Definition  Streicher_K : Streicher_K_.
End useless. (* everything's OK ! Don't care about goal :) *)
End  Test.
```

#### Coq Version
The Coq Proof Assistant, version 8.13.2 (April 2021)
compiled on Apr 15 2021 12:18:56 with OCaml 4.12.0
Linux Ubuntu

**Comment by ejgallego** [comment]
This is a duplicate of #13755 and has been fixed in `master` by #13790 , it was not selected to be backported due a minimal API change unfortunately.

Duplicate of #13755

**Comment by ejgallego** [comment]
Duplicate of #13755
